clusterman/draining/queue.py

```diff
@@ -357,6 +390,20 @@ def host_from_instance_id(
     except socket.error:
         logger.warning(f"Couldn't derive hostname from IP via DNS for {ip}")
         return None
+    try:
+        agent_id = instance_data[0]["PrivateDnsName"]
+    except KeyError:
+        logger.warning(f"No Private Dns Name found for {instance_id}")
+        return None
+    try:
+        for tag in instance_data[0]["Tags"]:
+            if "puppet:role::" in tag["Key"]:
```

Comment:
Really don't know what'd be the best approach here, cause I feel like entangling things with Puppet parameters is not the best thing, but at the same time wouldn't know if there's a much better way.
Would throwing the private DNS name, which should correspond to the k8s node ID, to the k8s APIs and then read the pool label out of that be much worse?

Reply:
I totally agree with you! I checked both and I was not happy none of them :) 
I think best way is adding new tag to the instance: https://sourcegraph.yelpcorp.com/misc/terraform-code/-/blob/modules/paasta_spot_asg_cluster/main.tf?subtree=true#L90:1

I am not sure whether adding new tag in the module will add tag in all existing instances.
It is still ok it will not add, we can add with small script. 

What do you think?

Reply:
I'm pretty sure that to do that you'd have to update the tag specification on the launch template for the ASGs, which then would only affect the newly launched instances.

Reply:
I am switching to DNS name now, then we can think about.

Reply:
@piax93 probably you said `KeyName` value which is different for mesos and k8s